src/Compilers/Core/Portable/SymbolDisplay/ObjectDisplayOptions.cs

```diff
@@ -40,5 +40,11 @@ internal enum ObjectDisplayOptions
         /// In Visual Basic, replace non-printable characters with calls to ChrW and vb* constants.
         /// </summary>
         EscapeNonPrintableCharacters = 1 << 4,
+
+        /// <summary>
+        /// In C# when specifying <see cref="IncludeCodePoints"/>, use a hexadecimal number for the code point.
+        /// In Visual Basic, use hexadecimal numbers in calls to ChrW.
+        /// </summary>
+        UseHexadecimalNumbersForCharacters = 1 << 5
```

Comment:
The existing `FormatPrimitive` methods use hexadecimal numbers for both numeric literals and character numbers based on the `useHexadecimalNumbers` parameter, whereas the existing implementation for constants of fields and default parameters inside `ToDisplayParts` always uses decimal numeric literals and hexadecimal numbers inside `ChrW`. To accomodate for this flexibility, I added this option.

I do find it slightly unfortunate that the new API won't have any option to behave the same way as constants displayed for symbols. This means that quick info will be inconsistent after all (showing hexadecimal for constant fields and default parameters but always decimal for results of expressions), but I don't know what can/should be done about it. I don't care that much about special characters in VB.

Reply:
Why can't this be controlled using SymbolDisplayOptions?

Reply:
Maybe. I'll wait for compiler review to see what their preference is.

Reply:
@CyrusNajmabadi That sounds like a good idea. We could maybe put this inside `SymbolDisplayMiscellaneousOptions` or make a new enum: `SymbolDisplayConstantValueOptions`. We could also add an overload of `FormatPrimitive` taking this. Although we'd need to figure out how to make it so that 2 of those options are on by default to match the existing default behavior for symbol display.

Reply:
> SymbolDisplayConstantValueOptions 

That's my preference.

> Alternatively we could negate them

That's my preference as well.  The quotes one, in particular, should be a 'NoQuotes' value.  By default, strings should be printed as they would be as literals (i.e. with quotes).  So you'd the option to opt out of that instead of needing to opt into it.

Reply:
@CyrusNajmabadi There's also the problem that for regular numeric literals, decimal should be the default (so there would be an option like UseHexadecimalNumbers) but for numbers for character values, the default should be the opposite to match current behavior. So then we'd need a strange option like DoNotUseHexadecimalNumbersForEscapedCharacters :(

Reply:
or maybe UseDecimalNumbersForCharacters?

Reply:
can UseDecimal be the default, and have UseHexa be something that someone needs to opt into?  Existing code that goes through FormatPrimitive will map their booleans to teh right symbol options.